junit-platform-suite-commons/src/main/java/org/junit/platform/suite/commons/SuiteLauncherDiscoveryRequestBuilder.java

```diff
@@ -201,6 +207,16 @@ public LauncherDiscoveryRequest build() {
 		return delegate.build();
 	}
 
+	private List<DiscoverySelector> selectMethods(SelectMethod... methods) {
+		List<DiscoverySelector> selectors = new ArrayList<>();
+
+		for (SelectMethod selectMethod : methods) {
+			selectors.add(DiscoverySelectors.selectMethod(selectMethod.returnType().getName(), selectMethod.name(),
+				selectMethod.parameterTypes()));
+		}
+		return selectors;
```

Comment:
For each class you should also add an entry to `this.selectedClassNames`. Otherwise the classes may be filtered out.

You should also write a test for this.

Reply:
It is possible that the name of the class is already added to selectedClassNames, in this case a class would appear twice, is that a bad thing?

Reply:
No. Because the patterns are used in an include filter duplicates aren't a problem. It would be better to use a set to store the selected class name though.